flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyMessage.java

```diff
@@ -235,14 +240,15 @@ void releaseBuffer() {
 		ByteBuf write(ByteBufAllocator allocator) throws IOException {
 			checkNotNull(buffer, "No buffer instance to serialize.");
 
-			int length = 16 + 4 + 1 + 4 + buffer.getSize();
+			int length = 16 + 4 + 4 + 1 + 4 + buffer.getSize();
```

Comment:
can you please add a comment here explaining reasons of the math?

Reply:
I think the lengths for the corresponding fields can be seen clearly in the below write method.